lisa/tools/gpu_drivers.py

```diff
@@ -0,0 +1,754 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+from abc import abstractmethod
+from pathlib import PurePosixPath
+from typing import Any, List, Optional, Type, Union
+
+from lisa.base_tools import Sed, Uname, Wget
+from lisa.executable import Tool
+from lisa.operating_system import (
+    CBLMariner,
+    CpuArchitecture,
+    Oracle,
+    Posix,
+    Redhat,
+    Ubuntu,
+)
+from lisa.tools.amdsmi import AmdSmi
+
+# Import tools directly from their modules to avoid circular import.
+# lisa.tools.__init__.py imports from this file (gpu_drivers.py), so we cannot
+# import from lisa.tools package directly. Instead, import from individual modules.
+from lisa.tools.df import Df
+from lisa.tools.echo import Echo
+from lisa.tools.mkdir import Mkdir
+from lisa.tools.modprobe import Modprobe
+from lisa.tools.nvidiasmi import NvidiaSmi
+from lisa.util import LisaException, MissingPackagesException, SkippedException
+
+
+class GpuDriver(Tool):
+    """
+    Virtual tool that wraps GPU-specific monitoring tools (NvidiaSmi, AmdSmi).
+
+    This tool should be created with a driver_class parameter specifying
+    which driver installer to use.
+
+    Example usage:
+        gpu_driver = node.tools.create(
+            GpuDriver,
+            driver_class=AmdGpuDriver
+        )
+    """
+
+    _driver_class: Type["GpuDriverInstaller"]
+    _smi_class: Type[Union[AmdSmi, NvidiaSmi]]
+
+    @property
+    def command(self) -> str:
+        return "echo"
+
+    @classmethod
+    def create(
+        cls,
+        node: Any,
+        driver_class: Type["GpuDriverInstaller"],
+    ) -> "GpuDriver":
+        """
+        Create a GpuDriver instance with the specified driver installer class.
+
+        Args:
+            node: The node to create the tool for
+            driver_class: The driver installer class to use
+                         (AmdGpuDriver, NvidiaCudaDriver, or NvidiaGridDriver)
+
+        Returns:
+            GpuDriver instance configured for the specified driver
+        """
+        from lisa.tools.amdsmi import AmdSmi
+        from lisa.tools.nvidiasmi import NvidiaSmi
+
+        instance = cls(node)
+        instance._driver_class = driver_class
+
+        # Determine monitoring tool class based on driver installer
+        if driver_class == AmdGpuDriver:
```

Comment:
define a class method `smi` for driver class, it doesn't need to hardcode types here.